plugin/pkg/admission/priority/admission.go

```diff
@@ -189,8 +206,12 @@ func (p *Plugin) admitPod(a admission.Attributes) error {
 			pod.Spec.PriorityClassName = pcName
 		} else {
 			pcName := pod.Spec.PriorityClassName
-			if !priorityClassPermittedInNamespace(pcName, a.GetNamespace()) {
-				return admission.NewForbidden(a, fmt.Errorf("pods with %v priorityClass is not permitted in %v namespace", pcName, a.GetNamespace()))
+			// If ResourceQuotaScopeSelectors is enabled, we should let pods with critical priorityClass to be created
+			// any namespace where administrator wants it to be created.
+			if !p.featureGates.resourceQuotaFeatureGateEnabled {
```

Comment:
@liggit, would this guard be good enough?

Reply:
that matches my expectations as long as that feature has the potential to be turned off